ARROW-14558 [R] clarify OOP system wording in the Arrow vignette (#11651)
apache/arrow: r/vignettes/arrow.Rmd

## Class structure and package conventions

> C++ is an object-oriented language, so the core logic of the Arrow library is encapsulated in classes and methods. In the R package, these classes are implemented using the `R6` object-oriented programming system, and most of them are exported from the namespace.

Reviewer: ```suggestion
C++ is an object-oriented language, so the core logic of the Arrow library is encapsulated in classes and methods. In the R package, these classes are implemented as `R6` classes, most of which are exported from the namespace.
```

Reviewer: Could also link to the R6 package website

Author: Should we also link to Hadley's R6 (sub)chapter in _Advanced R_? Or would anyone interested find that as it's not really a hidden resource?
